src/app/cli/src/init/client.ml

```diff
@@ -549,6 +549,68 @@ let send_payment_graphql =
          printf "Dispatched payment with ID %s\n"
            ((response#sendPayment)#payment |> unwrap_user_command)#id ))
 
+let sign_payment =
+  let open Command.Param in
+  let open Cli_lib.Arg_type in
+  let receiver_flag =
+    flag "--receiver" ~aliases:["receiver"]
+      ~doc:"PUBLICKEY Public key to which you want to send money"
+      (required public_key_compressed)
+  in
+  let amount_flag =
+    flag "--amount" ~aliases:["amount"]
+      ~doc:"VALUE Payment amount you want to send" (required txn_amount)
+  in
+  let token_flag =
+    flag "--token" ~aliases:["token"]
+      ~doc:"TOKEN_ID The ID of the token to transfer" (optional token_id)
+  in
+  let args =
+    Args.zip4 Cli_lib.Flag.signed_command_common receiver_flag amount_flag
+      token_flag
+  in
+  Command.async ~summary:"sign a payment to an address"
+    (Command.Param.map args
+       ~f:(fun (common, receiver_pk, amount, token_id) () ->
+            let {Cli_lib.Flag.sender; fee; nonce; memo} = common in
+            let fee_token = Token_id.default in
+            let fee_payer_pk = sender in
+            let valid_until = Unsigned.UInt32.max_int in
+            let nonce = Option.value ~default:Unsigned.UInt32.zero nonce in
+            let memo = Option.map ~f:Signed_command_memo.of_string memo
+                        |> Option.value ~default:Signed_command_memo.empty in
+            let common: Signed_command_payload.Common.t =
+              { fee; fee_token; fee_payer_pk; nonce; valid_until; memo } in
+            let body: Signed_command_payload.Body.t =
+              let token_id = Option.value ~default:Token_id.default token_id in
+              Signed_command_payload.Body.Payment
+              Payment_payload.Poly.
+              { source_pk=sender; receiver_pk; token_id; amount } in
+            let payload = Signed_command_payload.Poly.{ common; body } in
+            let logger = Logger.create () in
+            let%bind wallets =
+              let conf_dir = Mina_lib.Conf_dir.compute_conf_dir None in
+              let wallets_disk_location = conf_dir ^/ "wallets" in
+              Secrets.Wallets.load ~logger
+                ~disk_location:wallets_disk_location
+            in
+            let sk_file = Secrets.Wallets.get_path wallets sender in
```

Comment:
Can you allow the user to specify the secret key file directly, rather than a wallet path? Using `Secrets.Wallets.get_path` assumes that they've named their file according to our naming scheme within a directory, which again I don't think we want to depend upon.

Reply:
done!